extensions/2.0/Khronos/KHR_texture_procedurals/README.md

```diff
@@ -0,0 +1,1351 @@
+</super>
+
+
+#### Input Stream Binding
+
+Streams must be explicitly specified by a geometric procedural node for texture coordinates.
+
+There are two variants for these types of nodes:
+  1. Ones which specify the stream by number / index.
+  2. Ones which specify the stream by a string identifier.
+
+Only the first is supported. Others must be remapped to a numbered streams.
+
+The stream number is a zero-based index used to lookup the corresponding stream type in a bound mesh's `primitives` array. 
+
+For the texture coordinate example if the stream number is 1, the binding is to stream: `TEXCOORD_1` The mesh to check would be the one assigned to the material using the procedural graph.
+
+For colors, a procedural node would be used to specify a color stream. A value of 1 would imply a binding to color stream `COLOR_1`.
+
+<details>
+<summary>Example graph using color and texture coordinate streams</summary>
+
+```JSON
+{
+  "extensions": {
+    "KHR_texture_procedurals": {
+      "procedurals": [
+        {
+          "name": "nodegraph1",
+          "nodetype": "nodegraph",
+          "type": "color3",
+          "nodes": [
+            {
+              "name": "geomcolor_color3",
+              "nodetype": "geomcolor",
+              "type": "color3",
+              "inputs": [
+                {
+                  "name": "index",
+                  "nodetype": "input",
+                  "type": "integer",
+                  "value": 1
+                }
+              ],
+              "outputs": [
+                {
+                  "nodetype": "output",
+                  "name": "out",
+                  "type": "color3"
+                }
+              ]
+            },
+            {
+              "name": "texcoord_vector2",
+              "nodetype": "texcoord",
+              "type": "vector2",
+              "inputs": [
+                {
+                  "name": "index",
+                  "nodetype": "input",
+                  "type": "integer",
+                  "value": 1
+                }
+              ],
+              "outputs": [
+                {
+                  "nodetype": "output",
+                  "name": "out",
+                  "type": "vector2"
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    }
+  }
+}
+```
+
+<super>Figure: Note that the `value` for the `index` input is 1. This implies that the binding is to streams `TEXCOORD_1` and `COLOR_1` for nodes `a_texcoord_vector2` and `a_geomcolor_color3` respectively.</super>
+
+The corresponding MaterialX representation would look like this:
+
+```xml
+<nodegraph name="nodegraph1">
+  <geomcolor name="a_geomcolor_color3" type="color3">
+    <input name="index" type="integer" value="1" />
+  </geomcolor>
+  <texcoord name="a_texcoord_vector2" type="vector2">
+    <input name="index" type="integer" value="1" />
+  </texcoord>
+</nodegraph>
+```
+
+</details>
+
+
+### Procedural Definitions
```

Comment:
I propose a solution where the procedural image extension uses a set of already declared operators/functions.
These functions are what are available, if need for future additions that will need to be done in a separate extension.

The upside of having pre-declared operators is that its straightforward to do implementation testcases to verify if an implementation is compliant or not.
Ie, for each operator there are a number of testcases that provides input data and are used to validate the output.

The total amount of operators needed is not that large, probably around 20, maybe even less.

Reply:
For a  non complete comparison - have a look at what we specified in the Ikea procedural prototypes
This is not the complete declaration but it's enough to show the differences in design:

https://github.com/idlabs/procedural-image